plugins/tech-insights-backend/api-report.md

```diff
@@ -92,6 +113,7 @@ export type TechInsightsContext<
 > = {
   factChecker?: FactChecker<CheckType, CheckResultType>;
   persistenceContext: PersistenceContext;
+  factRetrieverEngine: FactRetrieverEngine;
```

Comment:
You really would like to expose just a slim interface here, right, not the entire implementation class? Maybe even something with just a subset of the engine capabilities. The current code seems to invite the possibility of creating engines of your own and running scheduleI() etc which wasn't the intent, right?

Reply:
Yeah, I think it makes sense to slim it down a little. We do want the possibility to reschedule the fact retrievers, tasks implementation being idempotent we can just add them as tasks again. Also the possibility to run a retriever manually with the already exposed method is a good to have. I'll wrap this into an interface and export that instead.

Reply:
I've modified the PR to expose only an interface of the engine.